### ISSUE TITLE
A question about the 2D-diffusion

### ISSUE THREAD

**Issue by kimianoorbakhsh** [lululxvi/deepxde#176]
Hi, Thanks for this awesome library.

I am trying to solve the 2D diffusion equation "du_t = 10e-5(du_yy + du_zz)" using DEEPXDE and my own data (therefore, I do not need DEEPXDE to generate data for me).

I wrote the following code, but my test metric is always equal to zero in the results. Could you please help me with that? 

I also have to mention that in the following code, the train_x is a np array of shape (N,3) where the first column represents y, the second column represents z and the third one represents t (time). Also, the train_y is a np array of shape (N,1) which represent the output u. test_x, test_y are defined the same.

Many thanks.

    def pde(x, y):
        D = tf.Variable(10e-5)
        dT_t = dde.grad.jacobian(y, x, j=2)
        dT_zz = dde.grad.hessian(y, x, j=1)
        dT_yy = dde.grad.hessian(y, x, j=0)
        
        return (dT_t - D * (dT_yy + dT_zz))

    geom = dde.geometry.geometry_2d.Rectangle((0,0), (40,40))
    timedomain = dde.geometry.TimeDomain(0, 0.003)
    geomtime = dde.geometry.GeometryXTime(geom, timedomain)

    data = dde.data.pde.TimePDE(geomtime,pde,[])
    data.train_x = train_x
    data.train_y = train_y
    data.test_x = test_x
    data.test_y = test_y

    layer_size = [3] + [32] * 4 + [1]
    activation = "tanh"
    initializer = "Glorot uniform"
    net = dde.maps.FNN(layer_size, activation, initializer)

    model = dde.Model(data, net)

    model.compile("adam", lr=0.001, metrics=["l2 relative error"])
    losshistory, train_state = model.train(epochs=1000)

    dde.saveplot(losshistory, train_state, issave=False, isplot=True)`

**Comment by lululxvi** [comment]
There are a few problems in your code:
- Use `D = 10e-5`. 
- You don't have BC and IC. The problem is not well-defined.
- The way of using `train_x`, `train_y` etc is not correct. Just let DeepXDE sample the points for you.
- Your PDE is not well scaled. It is better to scale you PDE (e.g., D, x, t and the solution) to scale 1. See "Q: I failed to train the network or get the right solution, e.g., the training loss is large." at FAQ.

**Comment by kimianoorbakhsh** [comment]
> There are a few problems in your code:
> 
> * Use `D = 10e-5`.
> * You don't have BC and IC. The problem is not well-defined.
> * The way of using `train_x`, `train_y` etc is not correct. Just let DeepXDE sample the points for you.
> * Your PDE is not well scaled. It is better to scale you PDE (e.g., D, x, t and the solution) to scale 1. See "Q: I failed to train the network or get the right solution, e.g., the training loss is large." at FAQ.

Thank you very much. For your third comment, how can I let DeepXDE sample the points but still use my own train and test data? I am doing research work and I need to run this on my own data.

**Comment by kimianoorbakhsh** [comment]
I also have another question, if I want to ask DeepXDE to create the data for me, I need to implement the solution as well. I mean like the "func" function in your example of 1d diffusion which was something like this:
     `def func(x):
        return np.sin(np.pi * x[:, 0:1]) * np.exp(-x[:, 1:]) `

Could you please tell me how can I implement this for 2d?

Thanks

**Comment by lululxvi** [comment]
- `anchors` is the correct way to use your points `train_x` for training. What is your error?
- You can ignore the solution argument. See the following example, after we train the network, we predict the solution for the points we care. Then you can compute whatever you need.
https://github.com/lululxvi/deepxde/blob/5aa23114050b180bcb99d650acf78144b16fac3b/examples/Burgers.py#L48

**Comment by kimianoorbakhsh** [comment]
![image](https://user-images.githubusercontent.com/45353778/102174308-62f11680-3eb2-11eb-9f2c-664588edaa26.png)

I get this error, and this is because I did not use train_y in my code...
Where should I add it?

**Comment by lululxvi** [comment]
`model.compile("adam", lr=0.001)`

**Comment by Naveen-Surya** [comment]
def func(x): return np.sin(np.pi * x[:, 0:1]) * np.exp(-x[:, 1:])

Could you please tell me how can I implement this for 2d?

**Comment by kimianoorbakhsh** [comment]
Hello, Thanks for your suggestions, I ran the model on my data, and the result was something like this (which sounds very good):
![image](https://user-images.githubusercontent.com/45353778/102495980-077a8080-408c-11eb-9388-2d2d5cb54520.png)
But when I use the "predict function" on my test set, the values from the model and ground truth are completely different:
![image](https://user-images.githubusercontent.com/45353778/102496138-44467780-408c-11eb-9008-8549352b6a9f.png)

@lululxvi  Could you please help me with this? (I also have to mention that I do not use any normalization in my training!!!)

**Comment by lululxvi** [comment]
@kimianoorbakhsh Your PDE is not well defined. You don't have IC and BC in the code.

**Comment by engsbk** [comment]
I have a question about the 1D diffusion_exactBC example:
why is the output transformed in this manner?
```

net.apply_output_transform(
        lambda x, y: x[:, 1:2] * (1 - x[:, 0:1] ** 2) * y + tf.sin(np.pi * x[:, 0:1])
    )
```

I mean why this specific formula?

**Comment by lululxvi** [comment]
Because the IC and BC satisfy u(x,t)=sin(pi x)
